client/man/ipa-getkeytab.1

```diff
@@ -124,10 +116,10 @@ against a FreeIPA server more recent than version 3.3. The user requesting the
 keytab must have access to the keys for this operation to succeed.
 .SH "EXAMPLES"
 Add and retrieve a keytab for the NFS service principal on
-the host foo.example.com and save it in the file /tmp/nfs.keytab and retrieve just the des\-cbc\-crc key.
+the host foo.example.com and save it in the file /tmp/nfs.keytab and retrieve just the aes256\-sha2 key.
 
 .nf
-   # ipa\-getkeytab \-p nfs/foo.example.com \-k /tmp/nfs.keytab \-e des\-cbc\-crc
+   # ipa\-getkeytab \-p nfs/foo.example.com \-k /tmp/nfs.keytab \-e aes\-sha2
```

Comment:
aes256-sha2 is not a valid key name.

IPA supports:

krbSupportedEncSaltTypes: aes256-cts:normal
krbSupportedEncSaltTypes: aes256-cts:special
krbSupportedEncSaltTypes: aes128-cts:normal
krbSupportedEncSaltTypes: aes128-cts:special
krbSupportedEncSaltTypes: camellia128-cts-cmac:normal
krbSupportedEncSaltTypes: camellia128-cts-cmac:special
krbSupportedEncSaltTypes: camellia256-cts-cmac:normal
krbSupportedEncSaltTypes: camellia256-cts-cmac:special

Reply:
Why doesn't IPA support the SHA2 enctypes?  Is this tracked somewhere?

Reply:
This is not IPA problem -- `ipa-getkeytab` reports default encryption types from MIT Kerberos library.

```
# ipa-getkeytab --permitted-enctypes
Supported encryption types:
AES-256 CTS mode with 96-bit SHA-1 HMAC
AES-256 CTS mode with 192-bit SHA-384 HMAC
Camellia-256 CTS mode with CMAC
AES-128 CTS mode with 96-bit SHA-1 HMAC
AES-128 CTS mode with 128-bit SHA-256 HMAC
Camellia-128 CTS mode with CMAC
```
This comes from the following snippet in `client/ipa-getkeytab.c`:
```
        if (ret == -1 && permitted_enctypes &&
            !(server || principal || keytab || quiet)) {
                krb5_enctype *ktypes;
                char enc[79]; /* fit std terminal or truncate */

                krberr = krb5_get_permitted_enctypes(krbctx, &ktypes);
                if (krberr) {
                        fprintf(stderr, _("No system preferred enctypes ?!\n"));
                        exit(1);
                }
                fprintf(stdout, _("Supported encryption types:\n"));
                for (i = 0; ktypes[i]; i++) {
                        krberr = krb5_enctype_to_string(ktypes[i], enc, 79);
                        if (krberr) {
                                fprintf(stderr, _("Warning: "
                                        "failed to convert type (#%d)\n"), i);
                                continue;
                        }
                        fprintf(stdout, "%s\n", enc);
                }
                ipa_krb5_free_ktypes(krbctx, ktypes);
                exit (0);
        }
```

Reply:
The permitted encrypts are hardcoded in IPA install via install/share/kerberos.ldif

Reply:
Trying to understand... are the sha1 and sha2 enctypes being grouped together for some reason?  What's the right way to request (using krb5 enctype names) a keytab with aes256-cts-hmac-sha384-192 key?  with aes128-cts-hmac-sha256-128?

Reply:
Like I said, they aren't enabled currently. The enctypes need to be added in LDAP. I'll open a separate PR on that.

Reply:
https://github.com/freeipa/freeipa/pull/3841

Reply:
Sorry @rcritten - missed your comment and was replying to @abbra.  Thanks for opening the PR!